libtiledbsoma/src/reindexer/reindexer.cc

```diff
@@ -78,18 +88,22 @@ void IntIndexer::map_locations(const int64_t* keys, int size, int threads) {
     }
     auto hsize = kh_size(hash_);
     LOG_DEBUG(fmt::format("[Re-indexer] khash size = {}", hsize));
-    tiledb_thread_pool_ = std::make_unique<tiledbsoma::ThreadPool>(threads);
+    if (thread_count_ > 1) {
+        tiledb_thread_pool_ = std::make_unique<tiledbsoma::ThreadPool>(threads);
+    }
 }
 
 void IntIndexer::lookup(const int64_t* keys, int64_t* results, int size) {
     if (size == 0) {
         return;
     }
-    LOG_DEBUG(fmt::format(
-        "Lookup with thread concurrency {} on data size {}",
-        tiledb_thread_pool_->concurrency_level(),
-        size));
-    if (tiledb_thread_pool_->concurrency_level() == 1) {
+    if (tiledb_thread_pool_) {
+        LOG_DEBUG(fmt::format(
+            "Lookup with thread concurrency {} on data size {}",
+            tiledb_thread_pool_->concurrency_level(),
+            size));
+    }
+    if (thread_count_ == 1) {
```

Comment:
storing both `tiledb_thead_pool` and `thread_count` seems redundant.  Would be simpler to just use the former, e.g.,

```
if (tiledb_thread_pool_ == nullptr) {
    // do the single thread loop
    return
}

// else do the multi-thread path
```

You could remove `thread_count`, and move the new logging (line 101) into the MT branch